lib/Tool/Authentication.php

```diff
@@ -122,37 +121,19 @@ protected static function refreshUser(TokenInterface $token, UserProvider $provi
     {
         $user = $token->getUser();
 
-        $userNotFoundByProvider = false;
-        $userClass = $user::class;
-
-        if (!$provider instanceof UserProviderInterface) {
-            throw new \InvalidArgumentException(sprintf('User provider "%s" must implement "%s".', get_debug_type($provider), UserProviderInterface::class));
-        }
```

Comment:
The `Pimcore\Bundle\AdminBundle\Security\User\UserProvider` implements the `UserProviderInterface`, not need to check it again.